platforms/core-runtime/gradle-cli-main/src/main/java/org/gradle/launcher/GradleMain.java

```diff
@@ -16,26 +16,13 @@
 
 package org.gradle.launcher;
 
-import org.gradle.api.internal.jvm.JavaVersionParser;
-import org.gradle.internal.jvm.SupportedJavaVersions;
-import org.gradle.internal.jvm.UnsupportedJavaRuntimeException;
 import org.gradle.launcher.bootstrap.ProcessBootstrap;
 
 /**
  * The main entrypoint to the Gradle CLI Client.
  */
 public class GradleMain {
     public static void main(String[] args) throws Exception {
-        int currentMajorJavaVersion = JavaVersionParser.parseCurrentMajorVersion();
-        if (currentMajorJavaVersion < SupportedJavaVersions.MINIMUM_CLIENT_JAVA_VERSION) {
-            System.err.printf(UnsupportedJavaRuntimeException.getIncompatibleJavaVersionForProcessErrorMessage(
```

Comment:
`UnsupportedJavaRuntimeException.getIncompatibleJavaVersionForProcessErrorMessage` is no longer called from outside the class. We can now inline that method and simplify `UnsupportedJavaRuntimeException` a bit